src/collection/class.js

```diff
@@ -1,8 +1,15 @@
 let Set = require('../set');
+let is = require('../is');
 
 let elesfn = ({
   classes: function( classes ){
-    classes = ( classes || '' ).match( /\S+/g ) || [];
+    if( is.array( classes ) ){
+      // use a copy of the array
+      classes = classes.slice(0);
```

Comment:
I don't think the copying is necessary.  I don't think we mutate this list in functions like `toggleClass()`, and whatever json is specified for an element at init is considered owned by that element.  So it would be cheaper to avoid copying, if possible.

Reply:
That makes sense. I didn't check to see if it gets mutated or not, but if I remember correctly, the array gets put into a Set pretty quickly anyway. I'll remove the copy.

Reply:
Just some background info: We try to avoid copying except where absolutely necessary for performance reasons.  We've run into cases where copying causes issues before.  

Copying can affect large graphs.  For example, something like this can be common:

```js
cy.elements().forEach( ele => { // this callback would have many array copies made
  ele.toggleClass([ 'foo', 'bar' ]);

  if( eleIsBaz ){
    ele.addClass(['baz']);
  }
  
 // maybe even more class calls
} ); 
```

It all adds up, so we try to keep the lib as lean as possible to give the dev who uses the lib more flexibility.